opennms-base-assembly/src/main/filtered/etc/report-templates/NodeAvailabilityReport.jrxml

```diff
@@ -46,15 +46,24 @@
 		</parameter>
 		<parameter name="DS_AVAILABILITY_WARNING" class="java.lang.Double" isForPrompting="false"/>
 		<parameter name="DS_AVAILABILITY_CRITICAL" class="java.lang.Double" isForPrompting="false"/>
+		<parameter name="TIME_RANGE_NUMBER" class="java.lang.Integer" isForPrompting="false">
+			<defaultValueExpression><![CDATA[7]]></defaultValueExpression>
+		</parameter>
 		<parameter name="DS_TIME_RANGE" class="java.lang.String" isForPrompting="false">
-			<defaultValueExpression><![CDATA["7 Days"]]></defaultValueExpression>
+			<defaultValueExpression><![CDATA[$P{TIME_RANGE_NUMBER} + " Days"]]></defaultValueExpression>
 		</parameter>
 		<parameter name="DS_START_TIME" class="java.util.Date">
-			<defaultValueExpression><![CDATA[new java.util.Date(new java.util.Date().getTime() - (86400 * 7 * 1000))]]></defaultValueExpression>
+			<defaultValueExpression><![CDATA[new java.util.Date(new java.util.Date().getTime() - (86400000L * $P{TIME_RANGE_NUMBER}))]]></defaultValueExpression>
 		</parameter>
 		<parameter name="DS_START_TIME_STRING" class="java.lang.String" isForPrompting="false">
 			<defaultValueExpression><![CDATA["yyyy-MM-d HH:mm:ss Z"]]></defaultValueExpression>
```

Comment:
I have a question: shouldn't this defaultValueExpression be like this instead? Right now, it's just the DateFormat string which doesn't seem right.
```
<![CDATA[new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format($P{DS_START_TIME})]]>
```